src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -235,180 +276,215 @@ protected override AuthenticationTicket AuthenticateCore()
+                if (_logger.IsEnabled(LogLevel.Debug))
                 {
-                    ProtocolMessage = openIdConnectMessage
-                };
+                    _logger.LogDebug(format: Resources.OIDCH_0001_MessageReceived, args: message.BuildRedirectUrl());
+                }
+
+                var messageReceivedNotification =
+                    new MessageReceivedNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions>(Context, Options)
+                    {
+                        ProtocolMessage = message
+                    };
 
                 await Options.Notifications.MessageReceived(messageReceivedNotification);
                 if (messageReceivedNotification.HandledResponse)
                 {
+                    _logger.LogInformation(message: Resources.OIDCH_0002_MessageReceivedNotificationHandledResponse);
                     return messageReceivedNotification.AuthenticationTicket;
                 }
 
                 if (messageReceivedNotification.Skipped)
                 {
+                    _logger.LogInformation(message: Resources.OIDCH_0003_MessageReceivedNotificationSkipped);
                     return null;
                 }
 
-                // runtime always adds state, if we don't find it OR we failed to 'unprotect' it this is not a message we
-                // should process.
-                AuthenticationProperties properties = GetPropertiesFromState(openIdConnectMessage.State);
-                if (properties == null)
+                // runtime always adds state, if we don't find it OR we failed to 'unprotect' it this is not a message we should process.
+                if (string.IsNullOrWhiteSpace(message.State))
                 {
-                    _logger.LogWarning("The state field is missing or invalid.");
+                    _logger.LogError(message: Resources.OIDCH_0004_MessageStateIsNullOrWhiteSpace);
                     return null;
                 }
 
-                // devs will need to hook AuthenticationFailedNotification to avoid having 'raw' runtime errors displayed to users.
-                if (!string.IsNullOrWhiteSpace(openIdConnectMessage.Error))
-                {
-                    throw new OpenIdConnectProtocolException(
-                        string.Format(CultureInfo.InvariantCulture,
-                                      openIdConnectMessage.Error,
-                                      Resources.Exception_OpenIdConnectMessageError, openIdConnectMessage.ErrorDescription ?? string.Empty, openIdConnectMessage.ErrorUri ?? string.Empty));
-                }
-
-                // code is only accepted with id_token, in this version, hence check for code is inside this if
-                // OpenIdConnect protocol allows a Code to be received without the id_token
-                if (string.IsNullOrWhiteSpace(openIdConnectMessage.IdToken))
+                AuthenticationProperties properties = GetPropertiesFromState(message.State);
```

Comment:
Sure. But on the other hand, it would make it compliant with the latest guidelines :smile: 

>The var keyword is to be used as much as the compiler will allow.

https://github.com/aspnet/Home/wiki/Engineering-guidelines#usage-of-the-var-keyword